### PR TITLE
Add support for Hive dialect GROUPING SETS.

### PR DESCRIPTION
**Pull request by qzchenwl**
[Hive dialect](https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C+Grouping+and+Rollup#EnhancedAggregation,Cube,GroupingandRollup-GROUPINGSETSclause) requires all columns appear in GROUPING SETS clause to be in GROUP BY clause.

```
SELECT a, b, SUM( c ) FROM tab1 GROUP BY a, b GROUPING SETS ( (a, b), a, b, ( ) )
```

### CODE REVIEW

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 1]
```diff
@@ -2734,6 +2734,19 @@ GroupByElement GroupByColumnReferences():
     <K_GROUP> <K_BY>
     (     LOOKAHEAD(2) (
               "(" ")" { groupBy.withUsingBrackets(true); }
+              (
```
Comment (wumpz) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:2737]: Could you more compress this production? There are three nearly identical copies of this inserted block.

File: src/test/java/net/sf/jsqlparser/statement/select/HiveTest.java [hunk 2]
```diff
@@ -46,4 +46,24 @@ public void testLeftSemiJoin() throws Exception {
...
+        String sql;
+        Statement statement;
+
+        sql = "SELECT\n"
+            + "    C1, C2, C3, MAX(Value)\n"
+            + "FROM\n"
+            + "    Sometable\n"
+            + "GROUP BY C1, C2, C3 GROUPING SETS ((C1, C2), (C1, C2, C3), ())";
+
+        statement = CCJSqlParserUtil.parse(sql);
```
Comment (wumpz) [inline src/test/java/net/sf/jsqlparser/statement/select/HiveTest.java:61]: This hole parsing process is done using  assertSqlCanBeParsedAndDeparsed. So you could skip it.

**Review by wumpz — CHANGES_REQUESTED** [review]

**Comment by wumpz** [comment]
I still do not get the difference to our current grouping sets variant. Could you explain more specific what you want to achieve?

**Comment by qzchenwl** [comment]
JsqlParser could not parse `SELECT a, b, SUM( c ) FROM tab1 GROUP BY a, b GROUPING SETS ( (a, b), a, b, ( ) )`, it's a legal Hive statement.

It's an equivalent of Presto's `SELECT a, b, SUM( c ) FROM tab1 GROUP BY GROUPING SETS ( (a, b), a, b, ( ) )`.

**Pull request merged**